JRuby.md

```diff
@@ -0,0 +1,117 @@
+## JRuby
+
+> The Ruby Programming Language on the JVM
+
+This document serves as a getting-started guide for those wishing to use
+[Sorbet](https://sorbet.org) with [JRuby](https://www.jruby.org/).
+
+The use of Sorbet on the JVM is not **officially** supported by the Sorbet
+development team however this document services as a getting started guide
+for those who wish to trot on a unsupported path.
+
+### Performance
+
+TBD
+
+### Gotchas
+
+#### RBI for Java Code
```

Comment:
Non-blocking request:

Please use the same nesting structure as `docs/internals.md`. Namely: Start with `#` for the title of the doc, and only use `##` / `###` for sections and subsections (and no sub-subsections, i.e., no `####`)